applications/crossbar/src/modules/cb_resource_selectors.erl

```diff
@@ -180,22 +147,56 @@ content_types_accepted_by_verb(Context, _) ->
 -spec validate(cb_context:context(), path_token(), path_token()) -> cb_context:context().
 -spec validate(cb_context:context(), path_token(), path_token(),  path_token(), path_token()) -> cb_context:context().
 validate(Context) ->
-    validate_rules(set_account_db(Context), cb_context:req_verb(Context)).
+    summary(set_selectors_db(Context), [], <<"resource_selectors/id_listing">>, 'false').
 
+validate(Context, ?RULES) ->
+    validate_rules(set_account_db(Context), cb_context:req_verb(Context));
 validate(Context, ?NAME) ->
-    validate_selector(set_selectors_db(Context));
+    summary(set_selectors_db(Context), [], <<"resource_selectors/name_listing">>, 'true');
 validate(Context, ?RESOURCE) ->
-    validate_resource(set_selectors_db(Context)).
+    summary(set_selectors_db(Context), [], <<"resource_selectors/resource_listing">>, 'true');
+validate(Context, UUID) ->
+    validate_doc(set_selectors_db(Context), UUID, cb_context:req_verb(Context)).
 
 validate(Context, ?NAME, SelectorName) ->
-    validate_selector(set_selectors_db(Context), SelectorName);
+    summary(set_selectors_db(Context), [SelectorName], <<"resource_selectors/name_resource_listing">>, 'true');
 validate(Context, ?RESOURCE, ResourceId) ->
-    validate_resource(set_selectors_db(Context), ResourceId).
+    summary(set_selectors_db(Context), [ResourceId], <<"resource_selectors/resource_name_listing">>, 'true').
 
 validate(Context, ?RESOURCE, ResourceId, ?NAME, SelectorName) ->
-    validate_resource_selector(set_selectors_db(Context), ResourceId, SelectorName, cb_context:req_verb(Context));
+    summary(set_selectors_db(Context), [ResourceId, SelectorName], <<"resource_selectors/resource_name_id_listing">>, 'false');
 validate(Context, ?NAME, SelectorName, ?RESOURCE, ResourceId) ->
-    validate_resource_selector(set_selectors_db(Context), ResourceId, SelectorName, cb_context:req_verb(Context)).
+    summary(set_selectors_db(Context), [ResourceId, SelectorName], <<"resource_selectors/resource_name_id_listing">>, 'false').
+
+-spec validate_rules(cb_context:context(), http_method()) -> cb_context:context().
+validate_rules(Context, ?HTTP_GET) ->
+    load_rules(Context);
+validate_rules(Context, ?HTTP_POST) ->
+    OnSuccess = fun on_successful_rules_validation/1,
+    cb_context:validate_request_data(<<"resource_selectors.rules">>, Context, OnSuccess).
+
+-spec validate_doc(cb_context:context(), path_token(), http_method()) -> cb_context:context().
+validate_doc(Context, UUID, ?HTTP_GET) ->
+    crossbar_doc:load(UUID, Context, ?TYPE_CHECK_OPTION(?DOC_PVT_TYPE));
+validate_doc(Context, UUID, ?HTTP_POST) ->
+    OnSuccess = fun(C) -> on_successful_doc_validation(C, UUID) end,
+    cb_context:validate_request_data(<<"resource_selector">>, Context, OnSuccess);
```

Comment:
There is no significant reasons for it. When I was working on it just forgot to rename.